File: Eclipse/w7cpframe/src/clsPfGravityLoad.java
//
// Copyright (c)2016 S C Harrison
// Refer to License.txt for terms and conditions of use.
//
import java.io.*;
import java.util.regex.Matcher;
import java.util.regex.Pattern;


    class clsPfGravityLoad
    {

        public int f_action = 0;
        public double load = 0;        //.. mass per unit length of a member load ..

        public void initialise()
        {
            f_action = 0;
            load = 0;
        }

        public void setValues(int ActionKey, double LoadMag)
        {
            f_action = ActionKey;
            load = LoadMag;
        }

        public String sprint()
        {
            String s;

            s = "";
            s = s + String.format("%6d",f_action);
            s = s + String.format("%12.4f",load);

            return s;
        }

        public void cprint()
        {
            System.out.println(sprint());
        }

        public void fprint(BufferedWriter fp)
        {
            try {
				fp.write(sprint());
			} catch (IOException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
        }

        public void sgetData(String s)
        {
            String[] dataflds = new String[10]; //(0 To 9);
            int i;


            System.out.println("sgetData ...");

            //Trim Trailing Spaces from input String
            String regExpTrimPattern = "^\\s+|\\s+$"; //trim trailing spaces          
            Pattern regExpPatternObj = Pattern.compile(regExpTrimPattern);
            Matcher regExpMatches = regExpPatternObj.matcher(s);
            String regResult = regExpMatches.replaceAll("");
            
            //Get Data fields from input String
            String regExpNumberPattern = "-?\\d+(?:[,.]\\d+)?";
            regExpPatternObj = Pattern.compile(regExpNumberPattern);
            regExpMatches = regExpPatternObj.matcher(regResult);
            i=0;
            while (regExpMatches.find()){
            	dataflds[i] = regExpMatches.group().trim();
            	i = i+1;
            }


            f_action = Integer.parseInt(dataflds[0]);
            load = Double.parseDouble(dataflds[1]);


            System.out.println("... sgetData");
        }



    } //class

File: VisualStudio/VBasic/w7cpframe/clsPfGravityLoad.vb
'
' Copyright (c)2016 S C Harrison
' Refer to License.txt for terms and conditions of use.
'
Option Explicit On

Public Class clsPfGravityLoad

#Const isUsingExcel = 0

    Public f_action As Byte
    Public load As Double        '.. mass per unit length of a member load ..

    Sub initialise()
        f_action = 0
        load = 0
    End Sub

    Sub setValues(ByVal ActionKey As Integer, ByVal LoadMag As Double)
        f_action = ActionKey
        load = LoadMag
    End Sub

    Function sprint() As String
        sprint = StrLPad(Format(f_action, "##0"), 6) & StrLPad(Format(load, "0.000"), 15)
    End Function

    Sub cprint()
        Console.WriteLine(sprint)
    End Sub

    Sub fprint(ByVal fp As Integer)
        Print(fp, sprint)
    End Sub

#If isUsingExcel = 1 Then

  Sub wbkPrint(dataRec As Range, RecNum As Byte)
    With dataRec
      .Offset(RecNum, 0).Value = f_action
      .Offset(RecNum, 1).Value = load
    End With
  End Sub

  Sub wbkRead(dataRec As Range, RecNum As Byte)
    With dataRec
      f_action = .Offset(RecNum, 1).Value
      load = .Offset(RecNum, 2).Value
    End With
  End Sub

#End If

End Class
